Original patch from: gentoo/src/patchsets/glibc/2.9/1160_all_glibc-2.8-nscd-one-fork.patch

-= BEGIN original header =-
only fork one to assist in stop-start-daemon assumptions about daemon behavior

http://bugs.gentoo.org/190785

-= END original header =-

---
 nscd/nscd.c |   12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

--- a/nscd/nscd.c
+++ b/nscd/nscd.c
@@ -179,6 +179,9 @@
       if (pid != 0)
 	exit (0);
 
+       if (write_pid (_PATH_NSCDPID) < 0)
+	dbg_log ("%s: %s", _PATH_NSCDPID, strerror (errno));
+
       int nullfd = open (_PATH_DEVNULL, O_RDWR);
       if (nullfd != -1)
 	{
@@ -228,12 +231,6 @@
 	for (i = min_close_fd; i < getdtablesize (); i++)
 	  close (i);
 
-      pid = fork ();
-      if (pid == -1)
-	error (EXIT_FAILURE, errno, _("cannot fork"));
-      if (pid != 0)
-	exit (0);
-
       setsid ();
 
       if (chdir ("/") != 0)
@@ -242,9 +239,6 @@
 
       openlog ("nscd", LOG_CONS | LOG_ODELAY, LOG_DAEMON);
 
-      if (write_pid (_PATH_NSCDPID) < 0)
-        dbg_log ("%s: %s", _PATH_NSCDPID, strerror (errno));
-
       if (!init_logfile ())
 	dbg_log (_("Could not create log file"));
 
